q2_types/feature_data/_format.py

```diff
@@ -142,142 +143,168 @@ def _validate_(self, level):
     'TSVTaxonomyDirectoryFormat', 'taxonomy.tsv', TSVTaxonomyFormat)
 
 
-class DNAFASTAFormat(model.TextFileFormat):
+class FASTAFormat(model.TextFileFormat):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.aligned = False
+        self.alphabet = None
+
     def _validate_(self, level):
-        FASTADNAValidator = re.compile(r'[ACGTURYKMSWBDHVN]+\r?\n?')
-        ValidationSet = frozenset(('A', 'C', 'G', 'T', 'U', 'R', 'Y', 'K', 'M',
-                                   'S', 'W', 'B', 'D', 'H', 'V', 'N'))
+        FASTAValidator, ValidationSet = _construct_validator_from_alphabet(
+            self.alphabet)
+        self._validate_FASTA(level, FASTAValidator, ValidationSet)
 
-        _validate_DNAFASTAFormats(self, FASTADNAValidator, ValidationSet,
-                                  level)
+    def _validate_FASTA(self, level, FASTAValidator=None, ValidationSet=None):
+        last_line_was_ID = False
+        ids = {}
 
+        seq_len = 0
+        prev_seq_len = 0
+        prev_seq_start_line = 0
 
-DNASequencesDirectoryFormat = model.SingleFileDirectoryFormat(
-    'DNASequencesDirectoryFormat', 'dna-sequences.fasta', DNAFASTAFormat)
+        level_map = {'min': 100, 'max': float('inf')}
+        max_lines = level_map[level]
 
+        with skbio.io.util.open(str(self), encoding='binary') as fh:
```

Comment:
why use skbio for this?

Reply:
Hmm, I copied that part from genome-sampler's `GISAIDDNAFASTAFormat` (assuming there was a reason for doing it this way)... Just tried to use `self.path.open()` and that also seems to work no problem so I'll revert to that.

Reply:
the original goal in GISAIDDNAFASTAFormat was to make it indifferent to compression. Unfortunately that didn't end up working anyhow